code/controllers/subsystems/economy.dm

```diff
@@ -27,48 +27,39 @@ SUBSYSTEM_DEF(economy)
 	var/paid_external = 0
 
 	// Departments pay to departments first
-	for(var/datum/money_account/A in department_accounts)
-		if(!A.employer)
-			continue
-
+	for(var/i in department_accounts)
+		var/datum/money_account/A = department_accounts[i]
 		var/datum/department/D = GLOB.all_departments[A.department_id]
-		var/datum/department/ED = GLOB.all_departments[A.employer]
-		var/datum/money_account/EA = get_account(ED.account_number)
+		var/datum/department/ED = GLOB.all_departments[D.funding_source]
 
-		if(D && (D.funding_type != FUNDING_NONE) && !A.wage_manual)
+		if(D && !A.wage_manual)
 			A.wage = D.get_total_budget()
 
 		var/amount_to_pay = A.debt + A.wage
-
 		if(amount_to_pay <= 0)
 			continue
 
-		switch(D.funding_type)
-			if(FUNDING_NONE)
-				continue
-
-			if(FUNDING_EXTERNAL)
-				deposit_to_account(A, D.funding_source, "Payroll Funding", "Hansa payroll system", amount_to_pay)
-				paid_external += amount_to_pay
-
-			if(FUNDING_INTERNAL)
-				if(amount_to_pay <= EA.money)
-					transfer_funds(EA, A, "Payroll Funding", "CEV Eris payroll system", amount_to_pay)
-					paid_internal += amount_to_pay
-					ED.total_debt -= A.debt
-					A.debt = 0
-				else
-					A.debt += A.wage
-					ED.total_debt += A.wage
-
+		if(!ED) // If no employer department found - payment is external
```

Comment:
Just to clear a doubt since I don't have a clear view on the whole system. That `if` won't allow people to create a fake department with a single high-wage employee to "steal" money from Hansa payroll system, right ?

Reply:
That only work if employer is present but not a department, that is only possible for roundstart accounts e.g. Moebius Med and Moebius Sci.